lib/galaxy/managers/users.py

```diff
@@ -279,13 +280,11 @@ def add_serializers(self):
             'create_time'   : self.serialize_date,
             'update_time'   : self.serialize_date,
             'is_admin'      : lambda i, k, **c: self.user_manager.is_admin(i),
-
             'preferences'   : lambda i, k, **c: self.user_manager.preferences(i),
-
             'total_disk_usage' : lambda i, k, **c: float(i.total_disk_usage),
+            'deleted_disk_usage' : lambda i, k, **c: int(i.calculate_deleted_disk_usage()),
```

Comment:
I'm concerned that this is far too expensive to calculate on the fly -- user serialization must be very fast since it's happening frequently.

Reply:
Good point. 
We potentially could use the same method as the total_disk_usage, which calculates and stores the number at the back sever, and displays the number dynamically with users' input. However,  my concern on this approach is that the displayed number might be quite different from the real one when the starting number is small and multiple delete operations occurs, which compromises users' experience. 
Another approach is setting up a pop-up modal view triggered exclusively by click on the quotameter. The deleted disk usage could be calculated and displayed there to user. Besides, the total disk usage could be recalculated and set there, too.
What do you think?

Reply:
@qiagu I think we probably need *both* of those things -- a cached calculation of the user's dataset details as well as a user-side (currently recalculations are admin-invoked and happen on user logout) recalc method in the interface.

Reply:
@dannon I like your suggestion a lot. Thanks.